tests/org.pitest.pitclipse.ui.tests/src/org/pitest/pitclipse/ui/behaviours/pageobjects/PitSummaryView.java

```diff
@@ -16,34 +16,100 @@
+import static org.junit.Assert.fail;
+
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.waits.DefaultCondition;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotBrowser;
+import org.pitest.pitclipse.ui.view.PitView;
 
 public class PitSummaryView {
+    private SWTWorkbenchBot bot;
+    private SWTBotBrowser browser;
+
+    public PitSummaryView(SWTWorkbenchBot bot) {
+        this.bot = bot;
+    }
 
-    private PitResultsView lastResults = null;
+    public String clickBack(String expectedUrl) {
+        showSummaryView();
+        bot.toolbarButton(PitView.BACK_BUTTON_TEXT).click();
+        bot.waitUntil(new BrowserLoadCondition(expectedUrl));
+        return getCurrentBrowserUrl();
+    }
 
-    public PitSummaryView() {
+    public String clickHome(String expectedUrl) {
+        showSummaryView();
+        bot.toolbarButton(PitView.HOME_BUTTON_TEXT).click();
+        bot.waitUntil(new BrowserLoadCondition(expectedUrl));
+        return getCurrentBrowserUrl();
     }
 
-    public void waitForUpdate() {
-        try {
-            lastResults = PitNotifier.INSTANCE.getResults();
-        } catch (InterruptedException e) {
-            throw new StepException(e);
+    public String clickForward(String expectedUrl) {
+        showSummaryView();
+        bot.toolbarButton(PitView.FORWARD_BUTTON_TEXT).click();
+        bot.waitUntil(new BrowserLoadCondition(expectedUrl));
+        return getCurrentBrowserUrl();
+    }
+
+    /**
+     * Shows the view, which allows the bot to find toolbar buttons
+     */
+    private void showSummaryView() {
+        bot.viewById(PitView.VIEW_ID).show();
+    }
+
+    public void getBrowserIfNotSet() {
+        if (browser == null) {
+            browser = bot.viewById(PitView.VIEW_ID).bot().browser();
+            if (browser == null) {
+                fail("Couldn't get browser of '" + PitView.VIEW_ID + "'");
+            }
         }
     }
 
-    public int getClassesTested() {
-        return lastResults.getClassesTested();
+    public String getCurrentBrowserUrl() {
+        getBrowserIfNotSet();
+        return browser.getUrl();
     }
 
-    public double getOverallCoverage() {
-        return lastResults.getTotalCoverage();
+    public String setBrowserUrl(String url) {
+        getBrowserIfNotSet();
+        browser.setUrl(url);
+        bot.waitUntil(new BrowserLoadCondition(url.replace(".html", "")));
+        return getCurrentBrowserUrl();
     }
 
-    public double getMutationCoverage() {
-        return lastResults.getMutationCoverage();
+    private class BrowserLoadCondition extends DefaultCondition {
+        private final String titleOfPage;
+        private String html;
+        public BrowserLoadCondition(String titleOfPage) {
+            getBrowserIfNotSet();
+            // give the browser time to change.
+            // To avoid false positives, if the page should not change
+            browser.waitForPageLoaded();
```

Comment:
Do you mean to move this in the method of `PitclipsePitSummaryViewTest.java`, if so I disagree?
The browser should wait here, because afterwards the code gets the Url, example from above:
`bot.waitUntil(new BrowserLoadCondition(url.replace(".html", "")));
return getCurrentBrowserUrl();`

Reply:
No, I meant in `BrowserLoadCondition::test`.

Current implementation could cause issues, e.g.:
```java
// waits for the page to load here
ICondition condition = new BrowserLoadCondition(url);

// but we change the page here
browser.setUrl(url);

// and thus we may test the page before it's been fully loaded
bot.waitUntil(condition);
```

That's unlikely, I agree, but I see no reason not to fix it, just in case.

Reply:
I agree.